src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,43 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.exception.EmptyDescriptionException;
+import duke.tasklist.TaskList;
+
+/**
+ * Represents a command to delete task.
+ */
+public class DeleteCommand extends Command {
+    private int taskIndex;
+
+    /**
+     * Creates a DeleteCommand.
+     *
+     * @param description Task to be deleted.
+     * @throws DukeException If description is empty or is of an invalid format.
+     */
+    public DeleteCommand(String description) throws DukeException {
+        super(description);
+
+        if (description == "") {
+            throw new EmptyDescriptionException("OOPS!!! The description cannot be empty.");
+        }
+        try {
+            int taskNumber = Integer.valueOf(description);
+            taskIndex = taskNumber - 1;
+        } catch (NumberFormatException e) {
+            throw new DukeException("Please input number after unmark");
+        }
```

Comment:
Errors that you may consider checking for:
1. Invalid number, and
2. Index out of bounds